utils/s2n_socket.c

```diff
@@ -37,6 +37,28 @@
     #define S2N_CORK_OFF    1
 #endif
 
+int s2n_socket_quickack(struct s2n_connection *conn)
+{
+#ifdef TCP_QUICKACK
+    if (!conn->managed_io) {
+        return 0;
+    }
+
+    int optval = 1;
+
+    struct s2n_socket_read_io_context *r_io_ctx = (struct s2n_socket_read_io_context *) conn->recv_io_context;
+    if (r_io_ctx->tcp_quickack_set) {
+        return 0;
+    }
+
+    /* Ignore the return value, if it fails it fails */
+    setsockopt(r_io_ctx->fd, IPPROTO_TCP, TCP_QUICKACK, &optval, sizeof(optval));
```

Comment:
I agree with not treating the call as an error and bombing out, but should we really set the next bit to 1 if this function didn't succeed?

Reply:
ah, good catch.  i'll update that.